examples/todo-list/src/main.rs

```diff
@@ -0,0 +1,92 @@
+use std::sync::Arc;
+
+use askama::Template;
+use flareon::forms::Form;
+use flareon::prelude::{Body, Error, FlareonApp, FlareonProject, Response, Route, StatusCode};
+use flareon::request::Request;
+use flareon::reverse;
+use tokio::sync::RwLock;
+
+#[derive(Debug, Clone)]
+struct TodoItem {
+    title: String,
+}
+
+#[derive(Debug, Template)]
+#[template(path = "index.html")]
+struct IndexTemplate<'a> {
+    request: &'a Request,
+    todo_items: Vec<TodoItem>,
+}
+
+static TODOS: RwLock<Vec<TodoItem>> = RwLock::const_new(Vec::new());
+
+async fn index(request: Request) -> Result<Response, Error> {
+    let todo_items = (*TODOS.read().await).clone();
+    let index_template = IndexTemplate {
+        request: &request,
+        todo_items,
+    };
+    let rendered = index_template.render().unwrap();
+
+    Ok(Response::new_html(
+        StatusCode::OK,
+        Body::fixed(rendered.as_bytes().to_vec()),
+    ))
+}
+
+#[derive(Debug, Form)]
+struct TodoForm {
+    #[form(opt(max_length = 100))]
+    title: String,
+}
+
+async fn add_todo(mut request: Request) -> Result<Response, Error> {
+    let todo_form = TodoForm::from_request(&mut request).await.unwrap();
+
+    {
+        let mut todos = TODOS.write().await;
+        todos.push(TodoItem {
+            title: todo_form.title,
+        });
+    }
+
+    Ok(reverse!(request, "index"))
+}
+
+async fn remove_todo(request: Request) -> Result<Response, Error> {
+    let todo_id = request.path_param("todo_id").expect("todo_id not found");
+    let todo_id = todo_id.parse::<usize>().expect("todo_id is not a number");
```

Comment:
It would probably be nice to have an ergonomic API to get path parameters with type safety (similar to `Form` I guess), but this is definitely out of scope of this PR.

Reply:
I think that the `fn(request) -> response)` was inspired by Django's approach, but I think I'd rather go with Rocket's /Axum's approach of variables as function arguments, I think those seem more convenient to use.